### DISCUSSION TITLE
Toolbox like winforms

### DISCUSSION THREAD

**Discussion by MasterThreadInf** [unoplatform/uno#11362] [Q&A]
Is it possible to get toolbox like we have for WinForms ?
So that we can work without interruption and without worrying about adding code in xaml . Also sometimes even if code is correct ,error appears which gets difficult to seperate real error from this error (for xaml). 

Thanking in adavance!

**Comment by francoistanguay** [comment]
It's clearly something we've been asked several times and that we're currently investigating. Make sure to follow our upcoming releases/previews and raise your hand if you want to provide early feedback on previews if we decide to ship something.

**Reply by MasterThreadInf** [reply]
Thank you for considering to start toolbox. I really appreciate that, it'll be definetely helpful. By the way, I was going through Uno Sample and I came across example name "UnoWTCDataGridSample" which works with old settings but for new structure (with net 6.0) where instead of shared class there is class library and where netstandard 2.0 is added. Sample example doesn't work in this new structure. May be it need packages that I don't know or I'm adding packages which aren't needed. Please check sample of both in link below.

Old : https://github.com/MasterThreadInf/UNODataGrid
New: https://github.com/MasterThreadInf/UnoDataGirdInNewStructure

**Reply by jeromelaban** [reply]
Our sample needs an update to the new project format, but you can find how to integrate the community toolkit [in our documentation](https://platform.uno/docs/articles/uno-community-toolkit.html?tabs=tabid-winui#referencing-the-windows-community-toolkit-from-a-cross-targeted-library).

**Reply by MasterThreadInf** [reply]
Yes sure it needs update. Thanks for link. Following error still comes.
sample.Wasm	: CS0234	The type or namespace name 'BindableMetadataProvider' does not exist in the namespace 'sample.Wasm' (are you missing an assembly reference?)

**Reply by jeromelaban** [reply]
Thanks for the update.

The error you're seeing is a result of another error, always make sure to take a look at the first error in the output log.

Based on the documentation, you'll need to make the community toolkit dependency conditional.

Remove this line:
```xml
<PackageReference Include="CommunityToolkit.WinUI.UI.Controls" Version="7.1.2" />
```

And use this instead:
```xml
  <ItemGroup Condition="'$(TargetFramework)' == 'net6.0-windows10.0.18362.0'">
    <PackageReference Include="CommunityToolkit.WinUI.UI.Controls" Version="7.1.2" />
  </ItemGroup>
  <ItemGroup Condition="'$(TargetFramework)' != 'net6.0-windows10.0.18362.0'">
    <PackageReference Include="Uno.CommunityToolkit.WinUI.UI.Controls" Version="7.1.100" />
  </ItemGroup>
```

**Reply by MasterThreadInf** [reply]
For windows:
![2023-03-09_09-10-40](https://user-images.githubusercontent.com/116153815/223960712-c65685a0-8672-4ec9-aaeb-8532bc45aa51.png)
yes I added that code when you refered me the link.

For wasm I used Include="Uno.CommunityToolkit.WinUI.UI.Controls" Version="7.1.100" and this error
CS0234 The type or namespace name 'BindableMetadataProvider' does not exist in the namespace appears
![image](https://user-images.githubusercontent.com/116153815/223949814-b94fcb4a-796b-4670-b349-772b59bf08cd.png)
and thats the first error for wasm. 

Currently I have comment out whole App.xaml.host.cs , will it have effect in anyway or it is only for hosting purpose?

**Reply by jeromelaban** [reply]
Take a look at the updated sample, it will be easier: https://github.com/unoplatform/Uno.Samples/tree/master/UI/UnoWCTDataGridSample
